GoogleForCreators/web-stories-wp
# Context Menu: Refactor right click menu provider

## Context



## Summary

Pulled out code from the massive [right click menu provider](https://github.com/GoogleForCreators/web-stories-wp/blob/e7cb5fe4dee961dbfc318dbc7b3c75b3373b6a02/packages/story-editor/src/app/rightClickMenu/provider.js) into smaller hooks.

This was one way to approach it, but happy to hear other ideas.

## Relevant Technical Choices

Note: All of these actions are tested in the karma tests already. May be redundant to test them at the unit level.

Paying special attention to the values that are pulled out of context. Not every change should cause updates in each hook 😉.

@merapi Has a different approach here: https://github.com/GoogleForCreators/web-stories-wp/blob/7a4914adabb6cc691f2d0eaccad5d9cc056cf6af/packages/story-editor/src/app/rightClickMenu/menus/emptyStateMenu.js. If y'all like setting up the objects in these hooks we can do that here instead. For now I've left that step in the provider to keep the hooks from getting too bloated.

## To-do

- [x] finish moving actions
- [x] add unit tests now that we can do that
- [ ] ~identify performance improvements (if any) using flame charts~ will be done in a later issue #10459 
    - Note: since these hooks all pour into the same provider, it may be hard to see any specific performance improvements. Especially with `useApplyStyle` (`useRichTextFormatting` makes the browser do a lot of work).

## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

Perform these steps for each element type and on the background:
1. Open the right click menu 
2. Test each action individually
3. All should act the same as before


## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #10212
